wsireg/reg_images/np_reg_image.py

```diff
@@ -5,6 +5,7 @@
 from wsireg.utils.im_utils import (
     std_prepro,
     guess_rgb,
+    read_array,
 )
 
 
```

Comment:
I wonder if in `__init__` of `NumpyRegImage` it wouldn't be better to type check the incoming array and convert to dask, or at the very least convert `zarr.Array` to `dask`. This will ensure dimensions can be squeezed and sliced, and dimension information (number of channels, RGB) can be appropriately detected. 

```python

if isinstance(image, zarr.array):
    self.image = da.from_zarr(image)
elif isinstance(image, np.ndarray):
    self.image = da.from_array(image)
else isinstance(dask_im, da.core.Array):
    self.image = image

# drop any unused dimensions
self.image = da.squeeze(self.image)

# may be better to catch this earlier, right now, none of the registration methods are suitable to 
# images with a z-axis, it's assumed the third axis is a channel axis
if self.image.ndim > 3:
    raise ValueError(f"wsireg only supports 2D images, provided image has shape {self.image.shape}")
```

Reply:
`da.from_zarr` is a convenience function to open a url to a zarr. I think we can just use `from_array` for _all_ array like objects (not just `np.ndarray`):

```python
if isinstance(da.core.Array, image):
  self.image = image
else:
  # https://docs.dask.org/en/latest/array-api.html#dask.array.from_array
  # Create dask array from something that looks like an array.
  # Input must have a .shape, .ndim, .dtype and support numpy-style slicing.
  self.image = da.from_array(image)
```

If you look at the `from_zarr` method, you see that it's just a helper for creating a `zarr.Array` and calling `from_array`: https://github.com/dask/dask/blob/4243359983d773905c6ff427401779e42b5186c2/dask/array/core.py#L3193-L3247